pkg/containermeta/store.go

```diff
@@ -0,0 +1,282 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package containermeta
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/status/health"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/retry"
+)
+
+const (
+	retryCollectorInterval = 30 * time.Second
+	pullCollectorInterval  = 5 * time.Second
+)
+
+// Store contains metadata for container workloads.
+type Store struct {
+	mu         sync.RWMutex
+	containers map[string]Container
```

Comment:
I'm thinking the store should be designed to be generic, with something like Kubernetes objects:

```
type Entity struct {
    EntityMeta
    Object interface{}
}

type EntityMeta struct {
	Kind        Kind
	ID          string
	Name        string
	Namespace   string
	Annotations map[string]string
	Labels      map[string]string

        State string // or Enum
        Source // or Sources maybe 🤔         
}
```

We could keep `Entity` as an interface with `GetMeta()` and `GetObject()` but not sure it's useful.

And the store:

```
type Store struct {
    ...
    objects map[string]Entity // key is kind + ns + id, or two levels with first a map per kind
}

func (s *Store) Subscribe(f Filter) { // Filter being some func, might be a bit complicated to implement it with great performances, so only watching a specific Kind could be enough.
}
```